CRM/Upgrade/Incremental/php/FiveSixtyOne.php

```diff
@@ -21,6 +21,23 @@
  */
 class CRM_Upgrade_Incremental_php_FiveSixtyOne extends CRM_Upgrade_Incremental_Base {
 
+  public function setPreUpgradeMessage(&$preUpgradeMessage, $rev, $currentVer = NULL): void {
+    if ($rev === '5.61.alpha1' && CRM_Core_DAO::singleValueQuery('SELECT id FROM civicrm_contribution_recur LIMIT 1')) {
+      $documentationUrl = 'https://docs.civicrm.org/dev/en/latest/financial/recurring-contributions/';
+      $documentationAnchor = 'target="_blank" href="' . htmlentities($documentationUrl) . '"';
+      $extensionUrl = 'https://docs.civicrm.org/dev/en/latest/financial/recurring-contributions/';
+      $extensionAnchor = 'target="_blank" href="' . htmlentities($extensionUrl) . '"';
+
+      $preUpgradeMessage .= '<p>' .
+        ts('This release contains a change to the behaviour of recurring contributions under some edge-case circumstances.')
+        . ' ' . ts('In recent releases the amount and currency on the recurring contribution record changed when the amount of a contribution against it was changed, indicating a change in future intent.')
+        . ' ' . ts('It is generally not possible to edit contributions linked to payment processors so for most sites this would never occur anyway.')
```

Comment:
I've just looked at 5 different payment processors (stripe, smartdebit, authnet, gocardless, bitpay) and they all allow editing contributions linked to payment processors.

Reply:
OK - what do they do to enable that? I only decided you couldn't edit them because when I did testing in this context it froze the amount field on contributions - I think this code does it & it seems kinda hard to change 

https://github.com/civicrm/civicrm-core/blob/3a54ba1af1ff2adb1ccaee334d34714ce61fa8b9/CRM/Contribute/Form/Contribution.php#L697-L702

Reply:
Ok, it only applies to one-off contributions. Contributions linked to recurring contributions don't allow editing.
Something like this instead?: _"You cannot edit contribution amounts through the UI if they are linked to a recurring contribution which is managed by a payment processor."_

Reply:
SO these are the copy-edits you want - https://github.com/civicrm/civicrm-core/pull/25776/commits/ec2d4969927f6bb380832283fe8cfff303d01b3e

Reply:
Thanks, looks good.